public/javascript/script.js

```diff
@@ -1,11 +1,126 @@
-function populateRequest(event, form, response) {
-  var method = document.getElementById('formz').elements
-  for (let things in method){
-    console.log(things.value)
+let request
+
+const populateRequest = () => {
+  const formElements = document.mainform.elements
+  let elementValuesObject = {}
+
+  for (let element of formElements){
+    elementValuesObject[element.name] = element.value
+  }
+
+  let requestMethodHost = `${elementValuesObject.method} / HTTP/1.1
+Host: ${elementValuesObject.host}
+`
+
+  let requestHeader = ``
+  if (elementValuesObject.header_key1) {
+    requestHeader = requestHeader +
+      `${elementValuesObject.header_key1}: ${elementValuesObject.header_value1}
+`
+  }
+
+  if (elementValuesObject.header_key2) {
+    requestHeader = requestHeader +
+      `${elementValuesObject.header_key2}: ${elementValuesObject.header_value2}
+`
+  }
+
+  if (elementValuesObject.header_key3) {
+    requestHeader = requestHeader +
+      `${elementValuesObject.header_key3}: ${elementValuesObject.header_value3}
+`
+  }
+
+  let requestQuery = ``
+
+  if (elementValuesObject.key1) {
```

Comment:
colapse the places you're repeating code into for loops
use `i` in place of the key value. something like:
```
if(elementValuesObject.key[i]) {
...
```